index.js

```diff
@@ -2984,37 +2984,55 @@
 
   //. ### Array
 
-  //# append :: a -> Array a -> Array a
+  //# append :: (Applicative f, Semigroup (f a)) => a -> f a -> f a
```

Comment:
this explanation might not be precise, correct but might be helpful:

Kind of `f` is `* -> *`.
in order for something to be `Semigroup` it should be of kind `*` for example we can substitute `f` with `Maybe`. `Maybe` is of kind `* -> *`, there can't be any **value** of that type (values are of kind `*`).

For something to be Bifunctor it's kind should be `* -> * -> * ` (i.e. should be parameterized by two types). In order something to be a `Monad` for example, it's kind should be `* -> * `.


If you take a look at monoid instance for [list](https://hackage.haskell.org/package/base-4.9.1.0/docs/src/GHC.Base.html):
```haskell
instance Monoid [a] where
        mempty  = []
        mappend = (++)
```
and [Monad](https://hackage.haskell.org/package/base-4.9.1.0/docs/src/GHC.Base.html)
```haskell
instance Monad []  where
    xs >>= f             = [y | x <- xs, y <- f x]
```

you can see here that Monad accepts something of kind `* -> *` and `[]` is of that kind, when `[a]` is of kind `*` and is used in Monoid instance.